components/eam/src/physics/cam/gw_common.F90

```diff
@@ -741,5 +745,1387 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+        	zbot=zbot(:,pver:1:-1)
+        	!get the layer index of pblh in layer for input in drag scheme
+                pblh_idx = pbuf_get_index('pblh')
+        	call pbuf_get_field(pbuf, pblh_idx, pblh)
+        	do i=1,pcols
+        	kpbl2d_in(i)=pblh_get_level_idx(zbot(i,:)-(state%phis(i)/gravit),pblh(i))
+        	end do
+                !
+                !get grid size for dx,dy
+                call grid_size(state,dx,dy)
+                !interface for orographic drag
+                !if (gwd_fd.eq.0) then
+                call gwdo_gsd(&
+                u3d=state%u(:ncol,pver:1:-1),v3d=state%v(:ncol,pver:1:-1),t3d=state%t(:ncol,pver:1:-1),&
+                qv3d=state%q(:ncol,pver:1:-1,1),p3d=state%pmid(:ncol,pver:1:-1),p3di=state%pint(:ncol,pver+1:1:-1),&
+                pi3d=state%exner(:ncol,pver:1:-1),z=zbot(:ncol,pver:1:-1),&
+                ncleff_ls=ncleff_ls,ncd_bl=ncd_bl,sncleff_ss=sncleff_ss,&
+                rublten=utgw(:ncol,pver:1:-1),rvblten=vtgw(:ncol,pver:1:-1),rthblten=ttgw(:ncol,pver:1:-1),&
+                dtaux3d_ls=dtaux3_ls(:ncol,pver:1:-1),dtauy3d_ls=dtauy3_ls(:ncol,pver:1:-1),&
+                dtaux3d_bl=dtaux3_bl(:ncol,pver:1:-1),dtauy3d_bl=dtauy3_bl(:ncol,pver:1:-1),&
+                dtaux3d_ss=dtaux3_ss(:ncol,pver:1:-1),dtauy3d_ss=dtauy3_ss(:ncol,pver:1:-1),&
+                dtaux3d_fd=dtaux3_fd(:ncol,pver:1:-1),dtauy3d_fd=dtauy3_fd(:ncol,pver:1:-1),&
+                dusfcg_ls=dusfc_ls(:ncol),dvsfcg_ls=dvsfc_ls(:ncol),&
+                dusfcg_bl=dusfc_bl(:ncol),dvsfcg_bl=dvsfc_bl(:ncol),&
+                dusfcg_ss=dusfc_ss(:ncol),dvsfcg_ss=dvsfc_ss(:ncol),&
+                dusfcg_fd=dusfc_fd(:ncol),dvsfcg_fd=dvsfc_fd(:ncol),&
+                xland=cam_in%landfrac,br=state%ribulk(:ncol),&
+                var2d=sgh(:ncol),oc12d=state%oc(:ncol),&
+                oa2d=state%oadir(:ncol,:),ol2d=state%ol(:ncol,:),&
+                znu=etamid(pver:1:-1),dz=dz(:ncol,pver:1:-1),pblh=pblh(:ncol),&
+                cp=cpair,g=gravit,rd=rair,rv=rh2o,ep1=zvir,pi=pi,bnvbg=nm(:ncol,pver:1:-1),&
+                dt=dtime,dx=dx,dy=dy,&
+                kpbl2d=kpbl2d_in,itimestep=dtime,gwd_opt=0,&
+                ids=1,ide=ncol,jds=0,jde=0,kds=1,kde=pver, &
+                ims=1,ime=ncol,jms=0,jme=0,kms=1,kme=pver, &
+                its=1,ite=ncol,jts=0,jte=0,kts=1,kte=pver, &
+                gwd_ls=gwd_ls,gwd_bl=gwd_bl,gwd_ss=gwd_ss,gwd_fd=gwd_fd )
+                !
+end subroutine gw_oro_interface
+!==========================================================================
+function pblh_get_level_idx(height_array,pblheight)
+implicit none
+real(8),intent(in),dimension(pver) :: height_array
+real(8),intent(in) :: pblheight
+integer :: pblh_get_level_idx
+       
+!local 
+integer :: i
+logical :: found
+
+pblh_get_level_idx = -1
+found=.False.
+
+do i = 1, pver
+        if((pblheight >= height_array(i+1).and.pblheight <height_array(i)))then
+                pblh_get_level_idx =  pver+1-i
```

Comment:
Here's another example of weird vertical alignment - why are you not indenting the all lines of the routine?